### DISCUSSION TITLE
Question about Hyades Services

### DISCUSSION THREAD

**Discussion by jmayer-lm** [DependencyTrack/hyades#1565] [Q&A]
Hello, 
We are working on a new FOSS contribution for Hyades. Before we begin implementing, we want to know more knowledge about Hyades' services. Looking at the diagram, we noticed that services sometimes send a message back to the API via the kafka broker. As well as, the services sometimes write straight to the database. What is the "norm" for services so we know what process to follow, when contributing back?

**Comment by nscuro** [comment]
Hey, only the API server *writes* to the database, all other services communicate via Kafka. Some of them perform *read* operations on the DB, for example to acquire the cluster ID, and [configuration](https://github.com/DependencyTrack/hyades/tree/main/quarkus-extensions/config-dependencytrack). The idea is that this allows read-only services to connect to a read replica, if the need arises.
